File: 240613/2n개 중에 n개의 숫자를 적절하게 고르기/choose-n-out-of-2n-properly.js
const fs = require('fs')
let input = fs.readFileSync(0).toString().trim().split("\n")

const n = parseInt(input[0])
const arr = input[1].split(" ").map(x => Number(x))

let ans = Infinity
function back(idx, sm, cnt, sm2) {
    
    if (idx >= arr.length) {
        if (cnt == n) {
            ans = Math.min(ans, Math.abs(sm-sm2))
        }
        return
    }


    back(idx+1, sm+arr[idx], cnt+1, sm2)
    back(idx+1, sm, cnt, sm2+arr[idx])
}

back(0, 0, 0, 0)
console.log(ans)

File: 240613/2n개 중에 n개의 숫자를 적절하게 고르기/choose-n-out-of-2n-properly.py
n = int(input())
arr = list(map(int, input().split()))

ans = float('inf')
def back(cnt, sm, sm2):
    global ans
    if cnt >= len(arr):
        if len(sm) == n:
            ans = min(ans, abs(sum(sm) - sum(sm2)))
        return

    back(cnt+1, sm + [arr[cnt]], sm2)
    back(cnt+1, sm, sm2 + [arr[cnt]])

back(0, [], [])
print(ans)